src/components/type-group/TypeGroup.js

```diff
@@ -0,0 +1,29 @@
+import React, { cloneElement } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './TypeGroup.css';
+
+const TypeGroup = ({ children, className, name, onSelect, selectedId }) => {
+    const clonedChildren = React.Children.map(children, (child) => cloneElement(child, {
+        groupName: name,
+        onSelect,
+        selected: child.props.id === selectedId,
+    }));
+
+    return (
+        <div className={ classNames(styles.group, className) }>
+            { clonedChildren }
+        </div>
+    );
+};
+
+TypeGroup.propTypes = {
+    children: PropTypes.node.isRequired,
```

Comment:
Shouldn’t this be element since we expect the children to be element(s)?

Reply:
It makes sense. But I need to change it to: 

```js
PropTypes.oneOfType([
   PropTypes.arrayOf(PropTypes.element),
   PropTypes.element
]).isRequired
```

instead. Does that make sense to you?

Reply:
Yep, it has to be like that.